### PR TITLE
Extensions: WPSC - Move cache test functionality from HOC to EasyTab component

### PR DESCRIPTION
**Pull request by donnapep**
This PR may have been made obsolete by #14016.

The cache test functionality is currently defined in the `wrapSettingsForm` higher order component. Since `EasyTab` is the only component that uses it, it makes sense to define the functionality there rather than inside the HOC. This PR makes that change.

## Testing

To test, click the _Test Cache_ button on the _Easy_ tab and ensure that test results are returned and that a success notice is displayed:

cache-test-success

### CODE REVIEW

**Comment by ockham** [comment]
> This PR may have been made obsolete by #14016.

That other PR still seems a bit plagued by bugs. In the meantime, I was thinking we could maybe dispatch `errorNotice` and `successNotice` actions (found in `state/notices/actions`) from [extensions/wp-super-cache/state/cache/actions.js](https://github.com/Automattic/wp-calypso/blob/f19b11901e3b354b62d6234642346c42604abb22/client/extensions/wp-super-cache/state/cache/actions.js)#testCache -- the main difference/advantage over this PR being that we don't need to manage visibility of notices from inside the component's lifecycle methods, which keeps it a bit leaner.

What do you think?

**Comment by donnapep** [comment]
> That other PR still seems a bit plagued by bugs. In the meantime, I was thinking we could maybe dispatch errorNotice and successNotice actions (found in state/notices/actions) from extensions/wp-super-cache/state/cache/actions.js#testCache -- the main difference/advantage over this PR being that we don't need to manage visibility of notices from inside the component's lifecycle methods, which keeps it a bit leaner.

@ockham Sounds like a good interim change. I'll update this PR later today. Are you still pursuing the other (better) alternative though?

**Comment by ockham** [comment]
>> That other PR still seems a bit plagued by bugs. In the meantime, I was thinking we could maybe dispatch errorNotice and successNotice actions (found in state/notices/actions) from extensions/wp-super-cache/state/cache/actions.js#testCache -- the main difference/advantage over this PR being that we don't need to manage visibility of notices from inside the component's lifecycle methods, which keeps it a bit leaner.
>
> @ockham Sounds like a good interim change. I'll update this PR later today.

Cool, thanks!

> Are you still pursuing the other (better) alternative though?

Yeah, since I think the underlying data-layer extensions-middleware is possibly broken and needs to be fixed so any extensions at all can use it. Or alternatively that I have a misconception of something that needs to be fixed 😄

**Comment by donnapep** [comment]
@ockham This is ready for another review. Pretty much made the same changes as for #13818. Cheers!

File: client/extensions/wp-super-cache/state/cache/selectors.js [hunk 1]
```diff
@@ -3,6 +3,10 @@
  */
 import { get } from 'lodash';
 
+function getCacheState( state ) {
+	return get( state, 'extensions.wpSuperCache.cache', {} );
```
Comment (ockham) [inline client/extensions/wp-super-cache/state/cache/selectors.js]: Bit of a nitpick, but I think you can drop the `get` here and access `state.extensions.wpSuperCache.cache` directly. Rationale -- If this selector is called, we can assume the wpSuperCache extension code and thus the `extensions.wpSuperCache` Redux substate to be present; and the `cache` reducer has an initial state which is set on startup.

Comment (ockham) [inline client/extensions/wp-super-cache/state/cache/selectors.js]: Fixed this in 3c94859.

**Pull request merged**